src/internal/JsonUtilities.ts

```diff
@@ -508,4 +509,52 @@ export function jsonToHistoryState(json: any): HistoryState {
     let createdAt = new Date(json["_created"]);
     delete json["_created"];
     return new HistoryState(json, createdAt);
+}
+
+export function timeRangeToQueryJson(obj: TimeRange): Object {
+    return {
+        type: "withinTimeRange",
+        lowerLimit: obj.from.getTime(),
+        upperLimit: obj.to.getTime()
+    }
+}
+
+export function groupedQueryToJson(obj: QueryGroupedHistoryStatesRequest): Object {
+    let retJson: any = {};
+    let queryJson: any = {
+        grouped: true,
+    };
+
+    let clauseJson: any;
+    if (!!obj.clause) {
+        clauseJson = {
+            type: "and",
+            clauses: [
+                queryClauseToJson(obj.clause),
+                timeRangeToQueryJson(obj.range),
+            ]
+        }
+    } else {
+        clauseJson = timeRangeToQueryJson(obj.range)
+    }
+    queryJson["clause"] = clauseJson;
+
+    retJson["query"] = queryJson;
+    if (!!obj.firmwareVersion) {
+        retJson["firmwareVersion"] = obj.firmwareVersion;
+    }
+    return retJson;
+}
+
+export function jsonToTimeRange(json: any): TimeRange {
+    return new TimeRange(new Date(json["from"]), new Date(json["to"]));
+}
+
+export function jsonToGroupedHistoryStates(json: any): GroupedHistoryStates {
+    let states: Array<HistoryState> = [];
+    let stateJsons = json["objects"];
+    for (var i = 0; i < stateJsons.length; i++) {
+        states.push(jsonToHistoryState(stateJsons[i]));
+    }
```

Comment:
> I think using iterator or forEach method is easier to understand than using index.

Using `for..in` is concise other than understandable. 

However, it is suggested not to use `for...in` for array. 
This the reason: http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea